### PR TITLE
__builtins: replace top-level imports with imported

### PR DESCRIPTION
**Pull request by huglovefan**
this way the modules are only imported for code that needs them

<del>this also includes a fix to make `__builtin_expect` use c_long since that requires adding an import, it's done using an alias to keep it lazy without having to type `imported!"..."` for every use</del> (edit: will do separately)

i checked and all the affected functions still work like before

```C
// C code
void va(int x, ...)
{ 
	va_list ap, ap2;
	__builtin_va_start(ap, x);
	int y = __builtin_va_arg(ap, int);
	__builtin_va_copy(ap2, ap);
	__builtin_va_end(ap2);
	__builtin_va_end(ap);
}
int main()
{
	float f = __builtin_fabsf(1.0f);
	double d = __builtin_fabs(1.0);
	long double ld = __builtin_fabsl(1.0L);

	long x = __builtin_expect(0, 0);

	va(0, 0);
}
```

this now doesn't list any `core.stdc.*` modules in the output of `dmd -v test.c`:

```C
// C code
int main()
{
	__builtin_assume(1);
}
```

### CODE REVIEW

**Comment by dkorpel** [comment]
> this way the modules are only imported for code that needs them

So is the goal to improve compile times?

> this also includes a fix to make __builtin_expect use c_long

I think this PR is small enough to let it slide, but in general, please don't mix refactors and bug fixes.

File: src/__builtins.di [hunk 1]
```diff
@@ -87,10 +83,13 @@ version (DigitalMars)
...
 
+    // Lazily imported on first use
+    private alias c_long = imported!"core.stdc.config".c_long;
+
     // Stub these out to no-ops
-    int   __builtin_constant_p(T)(T exp) { return 0; } // should be something like __traits(compiles, enum X = expr)
-    long  __builtin_expect()(long exp, long c) { return exp; }
-    void* __builtin_assume_aligned()(const void* p, size_t align_, ...) { return cast(void*)p; }
+    int    __builtin_constant_p(T)(T exp) { return 0; } // should be something like __traits(compiles, enum X = expr)
+    c_long __builtin_expect()(c_long exp, c_long c) { return exp; }
```
Comment (ljmf00) [inline src/__builtins.di:91]: `long` to `c_long` change should be in different PR.

**Comment by huglovefan** [comment]
re-pushed without the c_long fix, will do that separately

> So is the goal to improve compile times?

something like that (but the difference is probably small)

this is an automatically imported file so i thought it would make sense to avoid unnecessary work there

the functions using core.bitop already do this by importing the module inside template functions, this does it for the rest of the imports

this PR was 50% to see if this would be accepted (it's how i would write the file but unsure what others think), 50% to fix the c_long thing without adding a new top-level import for it (but i see core.stdc.config is already depended upon by one of the current top-level imports so adding it without this PR wouldn't be too bad)

**Comment by RazvanN7** [comment]
> this way the modules are only imported for code that needs the

How? Isn't the global alias going to instantiate the file anyway?

**Comment by huglovefan** [comment]
> How? Isn't the global alias going to instantiate the template anyway?

it works that way in .d files being compiled (listed on dmd command line) but not modules that are just imported like __builtins

in imported modules that aren't included in compilation, the template is instantiated only if the importing module uses the alias or a template that uses it, or if the module itself uses it outside a template/alias

i don't know if this is documented somewhere but this is how i observe it works

**Review by RazvanN7 — APPROVED** [review]
